hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/updatetokencustomfees/UpdateTokenCustomFeesDecoder.java

```diff
@@ -0,0 +1,206 @@
+import com.hedera.hapi.node.transaction.RoyaltyFee;
+import com.hedera.hapi.node.transaction.TransactionBody;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.AddressIdConverter;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.HtsCallAttempt;
+import com.hedera.node.app.service.contract.impl.utils.ConversionUtils;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.apache.commons.lang3.function.TriFunction;
+
+@Singleton
+public class UpdateTokenCustomFeesDecoder {
+
+    // Function selectors
+    protected static final String UPDATE_FUNGIBLE_TOKEN_CUSTOM_FEES_STRING =
+            "updateFungibleTokenCustomFees(address,(int64,address,bool,bool,address)[],(int64,int64,int64,int64,bool,address)[])";
+    protected static final String UPDATE_NON_FUNGIBLE_TOKEN_CUSTOM_FEES_STRING =
+            "updateNonFungibleTokenCustomFees(address,(int64,address,bool,bool,address)[],(int64,int64,int64,address,bool,address)[])";
+
+    // Tuple indexes
+    // The function call for updateFungibleTokenCustomFees and updateNonFungibleTokenCustomFees
+    // share the first two tuple token address and fixed fees
+    // but have different third tuple for fractional or royalty fees respectively
+    private static final int TOKEN_ADDRESS = 0;
+    private static final int FIXED_FEE = 1;
+    private static final int FRACTIONAL_FEE = 2;
+    private static final int ROYALTY_FEE = 2;
+
+    // Fixed Fee tuple indexes
+    private static final int FIXED_FEE_AMOUNT = 0;
+    private static final int FIXED_FEE_TOKEN_ID = 1;
+    private static final int FIXED_FEE_USE_HBARS_FOR_PAYMENT = 2;
+    private static final int FIXED_FEE_USE_CURRENT_TOKEN_FOR_PAYMENT = 3;
+    private static final int FIXED_FEE_FEE_COLLECTOR = 4;
+
+    // Fractional Fee tuple indexes
+    private static final int FRACTIONAL_FEE_NUMERATOR = 0;
+    private static final int FRACTIONAL_FEE_DENOMINATOR = 1;
+    private static final int FRACTIONAL_FEE_MIN_AMOUNT = 2;
+    private static final int FRACTIONAL_FEE_MAX_AMOUNT = 3;
+    private static final int FRACTIONAL_FEE_NET_OF_TRANSFERS = 4;
+    private static final int FRACTIONAL_FEE_FEE_COLLECTOR = 5;
+
+    // Royalty Fee tuple indexes
+    private static final int ROYALTY_FEE_NUMERATOR = 0;
+    private static final int ROYALTY_FEE_DENOMINATOR = 1;
+    private static final int ROYALTY_FALLBACK_FEE_AMOUNT = 2;
+    private static final int ROYALTY_FALLBACK_FEE_TOKEN_ID = 3;
+    private static final int ROYALTY_FALLBACK_FEE_USE_HBARS_FOR_PAYMENT = 4;
+    private static final int ROYALTY_FEE_FEE_COLLECTOR = 5;
+
+    @Inject
+    public UpdateTokenCustomFeesDecoder() {}
+
+    public @Nullable TransactionBody decodeUpdateFungibleTokenCustomFees(@NonNull final HtsCallAttempt attempt) {
```

Comment:
This isn't `@Nullable`, right? Would be good to have some javadoc here that ends with,
```
@throws IllegalArgumentException if the call cannot be decoded
```
Could be added in another PR.

Reply:
Added javadoc for both methods